tests/core-addon/integration/core_addon.js

```diff
@@ -30,15 +31,117 @@ describe("Selenium", function () {
     await this.driver.quit();
   });
 
-  it("should install the core add-on", async function () {
+  it("should un/enroll in Rally", async function () {

```

Comment:
I guess the implication though is that each test will need to depend on the previous test to get it into the proper state? For instance to test the study add-on page the core add-on needs to be installed and then the browser navigates to the proper page.

I'll try it and see how it feels to write tests like this, if I am missing something lmk. Having the environment reset for each test feels much cleaner but we'd have more repetition in the tests (which we could move to helper functions I suppose) and it'd add more time for each test, but it's pretty fast overall so maybe not as big of a deal as I am worrying it is.

Reply:
> 
> 
> I guess the implication though is that each test will need to depend on the previous test to get it into the proper state? For instance to test the study add-on page the core add-on needs to be installed and then the browser navigates to the proper page.

Yes, I'm afraid that's the implication :(

> I'll try it and see how it feels to write tests like this, if I am missing something lmk. Having the environment reset for each test feels much cleaner but we'd have more repetition in the tests (which we could move to helper functions I suppose) and it'd add more time for each test run, but it's pretty fast overall so maybe not as big of a deal as I am worrying it is.

It's probably better to go as you suggested: one test and then we split off helper functions and share them. It feels a bit cleaner as tests don't depend on the state of other tests. In the long run, this will reduce potential intermittent nightmares!

Reply:
OK let's discuss this further in issue #229, I think it'll help us later if we make the right division between these tests :) Today I took a stab at moving the existing tests to helper functions but I'm not super happy with it yet.

Reply:
Given other comments here, it sounds like we're moving forward with this but will keep working on ways to break this test up. I suspect these will come up naturally as we try to add more tests.